src/language_features/references.rs

```diff
@@ -43,46 +44,73 @@ pub fn editor_references(
         ReferencesResponse::Array(locations) => Some(locations),
         ReferencesResponse::None => None,
     } {
+        // Sort locations by (filename, line)
+        let mut locations = locations.to_vec();
```

Comment:
I think this could be replaced by just putting `mut` into pattern match: `if let Some(mut locations) = match result {`

Reply:
Okay, I'll adapt this